sdk/metricsadvisor/ai-metrics-advisor/review/ai-metrics-advisor.api.md

```diff
@@ -1022,6 +1022,13 @@ export type PostgreSqlDataFeedSource = {
     authenticationType: "Basic";
 };
 
+// @public
+export interface RestResponse {
```

Comment:
@xirzec, @jeremymeng, @sarangan12 I would imagine many of our other packages requiring to define and export this interface. Would it make sense to have this in core-client/core-rest-pipeline?

Reply:
Not a lot of our package return RestResponse. I did a search and only find a couple 
```
anomalydetector\ai-anomaly-detector\review\ai-anomaly-detector.api.md
communication\communication-short-codes\review\communication-short-codes.api.md
core\core-http\review\core-http.api.md
digitaltwins\digital-twins-core\review\digital-twins-core.api.md
metricsadvisor\ai-metrics-advisor\review\ai-metrics-advisor.api.md
quantum\quantum-jobs\review\quantum-jobs.api.md
```
Some are generated. Other hand-crafted ones are communication-short-codes (we probably already asked them to use corev2) and digital-twins-core?  Looks like local copy should be good.

Another thing is that `FullOperationResponse` is not 100% compatible with `HttpOperationResponse`.  For example, the http headers type is different.  We may have to re-define.

Reply:
> Another thing is that FullOperationResponse is not 100% compatible with HttpOperationResponse. For example, the http headers type is different. We may have to re-define.

Going to add this to #15916 for posterity

Reply:
> Not a lot of our package return RestResponse

`HttpResponse` is used a lot though so it makes sense to have it and some transform helper at a common place.

Reply:
> HttpResponse object is already defined in core-client package.

@sarangan12 is it? I don't see it in core-client.api.md

Reply:
I saw one in core-client-rest

Reply:
Ah.. It is in `core-client-rest` package. In such a case, I agree we could define a common one in core-client package.

Reply:
Ok, if we are agreeing that we do not need RestResponse in any common code, I don't have any other comments in this PR. @sarangan12, Can you please review this PR?  Please also update the wiki your wrote on migration to include a section on how to deal with the case when the _response is part of the public API like it is for Metrics Advisor here.

Reply:
@ramya-rao-a I have already left one comment about the best practice below. Regarding the `_response`, I believe it is already included at https://github.com/Azure/autorest.typescript/wiki/%60core-http%60-dependency-migration-to-%60core-client%60-%60core-rest-pipeline%60#change-to-the-_response-property Do you like any additional information?

Reply:
@sarangan12 The section on _response you linked to talks about how to update internal usage of _response. I was referring to the case where the _response is part of the public surface API like in Metrics Advisor which would then need someone to explicitly define `RestResponse` on their own.

@KarishmaGhiya Can you address the other comment @sarangan12 has on this PR so that we can close this out?

Reply:
@ramya-rao-a  Will do